### PR TITLE
MVP2-385 Add bet list filter routes

### PR DESCRIPTION
**Pull request by fernandotimm**

### CODE REVIEW

File: controllers/bets-controller.js [hunk 1]
```diff
@@ -18,9 +18,39 @@ const betService = require('../services/bet-service');
...
 const { isAdmin } = require('../helper');
+const { calculateAllBetsStatus } = require('../services/event-service');
 
 const WFAIR = new Erc20('WFAIR');
 
+const listBets = async (req, res, next) => {
+  // Validating User Inputs
+  const errors = validationResult(req);
+  if (!errors.isEmpty()) {
+    return next(new ErrorHandler(422, 'Invalid input passed, please check it'));
```
Comment (SamirHodzic) [inline controllers/bets-controller.js:29]: Please change message here and use one from error, due on this route you don't have any validations related to input - only `checkAdmin`.

File: controllers/bets-controller.js [hunk 2]
```diff
@@ -18,9 +18,39 @@ const betService = require('../services/bet-service');
...
+  }
+  let q = {}
```
Comment (SamirHodzic) [inline controllers/bets-controller.js:31]: Not sure why this assignment?

File: controllers/bets-controller.js [hunk 3]
```diff
@@ -18,9 +18,39 @@ const betService = require('../services/bet-service');
...
+  const betList = await betService.filterBets(
+    type,
+    category,
+    count,
+    page,
+    sortby,
+    searchQuery,
+  );
+
+  res.status(201).json(betList);
```
Comment (SamirHodzic) [inline controllers/bets-controller.js:51]: `201` is not proper for this.

**Review by SamirHodzic — CHANGES_REQUESTED** [review]

File: controllers/bets-controller.js (hunk 1 above)
Comment (fernandotimm) [reply controllers/bets-controller.js:29]: I'll remove the validation check as this function is only intended to list the Bets. 
For now there's no need even for checking _isAdmin_ as far as I can see.
I'll add a try catch block around and log errors with the logger.

File: controllers/bets-controller.js (hunk 2 above)
Comment (fernandotimm) [reply controllers/bets-controller.js:31]: It was a mistake. I'll remove it.

File: controllers/bets-controller.js (hunk 3 above)
Comment (fernandotimm) [reply controllers/bets-controller.js:51]: You are right. I'll use 200

**Comment by fernandotimm** [comment]
@gmussi @SamirHodzic I'm waiting for the approval of this PR to push the changes on the front-end. Thanks!

File: controllers/bets-controller.js [hunk 4]
```diff
@@ -18,9 +18,45 @@ const betService = require('../services/bet-service');
...
 const WFAIR = new Erc20('WFAIR');
 
+const listBets = async (req, res, next) => {
+  try {
+    const betList = await betService.listBets();
+    return res.status(200).json(calculateAllBetsStatus(betList));
+
+  } catch (err) {
+    logger.error(err);
+    return next(res.status(422).send(err));
```
Comment (SamirHodzic) [inline controllers/bets-controller.js:33]: No need for return as this is final line that is executed in catch block.

**Review by SamirHodzic — APPROVED** [review]

File: controllers/bets-controller.js (hunk 4 above)
Comment (fernandotimm) [reply controllers/bets-controller.js]: thanks. I already removed the `return` and pushed a new commit.

**Pull request merged**